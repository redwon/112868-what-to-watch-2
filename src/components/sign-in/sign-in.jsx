import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {Operations} from '../../reducer/user/user';

import Header from '../header/header';
import Footer from '../footer/footer';

const SignIn = ({onLogin}) => {
  let email = ``;
  let password = ``;

  const emailHandler = (evt) => {
    email = evt.target.value;
  };

  const passwordHandler = (evt) => {
    password = evt.target.value;
  };

  const submitHandler = (evt) => {
    evt.preventDefault();
    onLogin(email, password);
  };

  return (
    <div className="user-page">
      <Header className="user-page__head" isHideUserBlock={true}>
        <h1 className="page-title user-page__title">Sign in</h1>
      </Header>
      <div className="sign-in user-page__content">
        <form className="sign-in__form" onSubmit={submitHandler}>
          <div className="sign-in__fields">
            <div className="sign-in__field">
              <input
                className="sign-in__input"
                type="email"
                placeholder="Email address"
                name="user-email"
                id="user-email"
                onChange={emailHandler}
              />
              <label className="sign-in__label visually-hidden" htmlFor="user-email">
                Email address
              </label>
            </div>

            <div className="sign-in__field">
              <input
                className="sign-in__input"
                type="password"
                placeholder="Password"
                name="user-password"
                id="user-password"
                onChange={passwordHandler}
              />
              <label className="sign-in__label visually-hidden" htmlFor="user-password">
                Password
              </label>
            </div>
          </div>
          <div className="sign-in__submit">
            <button className="sign-in__btn" type="submit">
              Sign in
            </button>
          </div>
        </form>
      </div>

      <Footer />
    </div>
  );
};

SignIn.propTypes = {
  onLogin: PropTypes.func
};

const mapDispatchToProps = (dispatch) => ({
  onLogin: (email, password) => {
    dispatch(Operations.login(email, password));
  }
});

export {SignIn};

export default connect(null, mapDispatchToProps)(SignIn);
