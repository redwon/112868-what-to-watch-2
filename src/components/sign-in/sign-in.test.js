import React from 'react';
import renderer from 'react-test-renderer';

import {SignIn} from './sign-in';

it(`renders correctly`, () => {
  const tree = renderer
    .create(<SignIn
      onLogin={jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
