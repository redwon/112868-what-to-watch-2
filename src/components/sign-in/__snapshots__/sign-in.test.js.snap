// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user-page"
>
  Header
  <div
    className="sign-in user-page__content"
  >
    <form
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field "
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="email"
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field "
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="password"
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 
        2019
         What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
